WcaOnRails/app/models/competition.rb

```diff
@@ -1005,7 +1005,7 @@ def user_can_view?(user)
   end
 
   def user_can_view_results?(user)
-    results_posted? || (user&.can_admin_results? && !results.empty?)
+    (results_posted? || user&.can_admin_results?) && !results.empty?
```

Comment:
That condition seems to make much more sense than before to me.
I can't recall any good reason why we'd have done it the way, so I'm going to assume we just made a silly mistake here.